modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointManager.java

```diff
@@ -0,0 +1,214 @@
+import org.apache.ignite.internal.pagememory.configuration.schema.PageMemoryCheckpointConfiguration;
+import org.apache.ignite.internal.pagememory.configuration.schema.PageMemoryCheckpointView;
+import org.apache.ignite.internal.pagememory.persistence.PageMemoryImpl;
+import org.apache.ignite.internal.pagememory.persistence.store.FilePageStoreManager;
+import org.apache.ignite.internal.util.IgniteUtils;
+import org.apache.ignite.internal.util.worker.IgniteWorkerListener;
+import org.apache.ignite.lang.IgniteInternalCheckedException;
+import org.apache.ignite.lang.IgniteLogger;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Main class to abstract checkpoint-related processes and actions and hide them from higher-level components.
+ *
+ * <p>Implements default checkpointing algorithm which is sharp checkpoint but can be replaced by other implementations if needed.
+ *
+ * <p>Represents only an intermediate step in refactoring of checkpointing component and may change in the future.
+ *
+ * <p>This checkpoint ensures that all pages marked as dirty under {@link #checkpointTimeoutLock} will be consistently saved to disk.
+ *
+ * <p>Configuration of this checkpoint allows the following:
+ * <ul>
+ *     <li>Collecting all pages from configured dataRegions which was marked as dirty under {@link #checkpointTimeoutLock}.</li>
+ *     <li>Marking the start of checkpoint on disk.</li>
+ *     <li>Notifying the subscribers of different checkpoint states through {@link CheckpointListener}.</li>
+ *     <li>Synchronizing collected pages with disk using {@link FilePageStoreManager}.</li>
+ * </ul>
+ */
+public class CheckpointManager implements IgniteComponent {
+    /** Checkpoint worker. */
+    private final Checkpointer checkpointer;
+
+    /** Main checkpoint steps. */
+    private final CheckpointWorkflow checkpointWorkflow;
+
+    /** Checkpoint markers storage which mark the start and end of each checkpoint. */
+    private final CheckpointMarkersStorage checkpointMarkersStorage;
+
+    /** Timeout checkpoint lock which should be used while write to memory happened. */
+    private final CheckpointTimeoutLock checkpointTimeoutLock;
+
+    /** Checkpoint page writer factory. */
+    private final CheckpointPagesWriterFactory checkpointPagesWriterFactory;
+
+    /**
+     * Constructor.
+     *
+     * @param logger Logger producer.
+     * @param igniteInstanceName Ignite instance name.
+     * @param checkpointConfig Checkpoint configuration.
+     * @param workerListener Listener for life-cycle checkpoint worker events.
+     * @param longJvmPauseDetector Long JVM pause detector.
+     * @param filePageStoreManager File page store manager.
+     * @param dataRegions Data regions.
+     * @param storagePath Storage path.
+     * @param pageSize Page size in bytes.
+     * @throws IgniteInternalCheckedException If failed.
+     */
+    public CheckpointManager(
+            Function<Class<?>, IgniteLogger> logger,
```

Comment:
Is there another way to do this? What was your intention?
AFAIK, we have only one entrance into getting logger instance, why bother?

Reply:
We should remember to rewrite this part anyways. I believe that we have plans to store information like "instance name" in loggers thread-local collection, and it would be convenient to get it from there. Design is not yet completed

Reply:
And why would you call this parameter "logger", makes no sense!

Reply:
How, in your opinion, should the logger be transferred to the internal components of the manager? Renamed the variable.

Reply:
I don't understand what you mean. Component can call IgniteLogger.forClass, can it?

Reply:
Yes, you are right.

Fix it.